server/src/main/java/org/elasticsearch/index/query/IntervalsSourceProvider.java

```diff
@@ -613,10 +625,10 @@ public void extractFields(Set<String> fields) {
         public boolean equals(Object o) {
             if (this == o) return true;
             if (o == null || getClass() != o.getClass()) return false;
-            Prefix prefix = (Prefix) o;
-            return Objects.equals(pattern, prefix.prefix) &&
-                Objects.equals(analyzer, prefix.analyzer) &&
-                Objects.equals(useField, prefix.useField);
+            Wildcard wildcard = (Wildcard) o;
```

Comment:
great catch, makes sense but maybe include this in a follow up PR adding the wildcard tests so we have code change coming with test.